barman/compression.py

```diff
@@ -638,6 +641,11 @@ def validate(self, pg_server_version, remote_status):
                 "backup_compression_level %d unsupported by "
                 "pg_basebackup compression %s" % (self.config.level, self.config.type)
             )
+        if self.config.workers is not None and (type(self.config.workers) is not int or self.config.workers < 0):
```

Comment:
Should we also validate that workers is not set in the validation for gzip and lz4?

Reply:
That was the idea. but I was counting in actually testing it before going further.

Reply:
This check now exists in both lz4 and gzip contexts